src/plugins/vis_types/vislib/public/pie.ts

```diff
@@ -11,16 +11,23 @@ import { VisTypeDefinition } from '../../../visualizations/public';
 import { CommonVislibParams } from './types';
 import { toExpressionAst } from './to_ast_pie';
 
-export interface PieVisParams extends CommonVislibParams {
+export enum LegendDisplay {
+  SHOW = 'show',
+  HIDE = 'hide',
+  DEFAULT = 'default',
+}
+
+export type PieVisParams = Omit<CommonVislibParams, 'addLegend'> & {
   type: 'pie';
   isDonut: boolean;
   labels: {
     show: boolean;
     values: boolean;
-    last_level: boolean;
+    lastLevel: boolean;
     truncate: number | null;
   };
-}
+  legendDisplay: LegendDisplay;
```

Comment:
While testing this I realized you can't turn the legend on in vislib pie - the button in the bottom left is missing. This is also happening in 8.0. Just a note, I don't think we have to solve on this PR